controllers/rhmi/bootstrapReconciler.go

```diff
@@ -476,6 +476,13 @@ func (r *Reconciler) reconcileOauthSecrets(ctx context.Context, serverClient k8s
 			oauthClientSecrets.Data = map[string][]byte{}
 		}
 		for _, product := range productsList {
+			// ugly check the first char is the same as the next three to remove duplicate string if it is, would like a better way to do this
+			// this can be removed again after a successful upgrade using this version
+			buf := oauthClientSecrets.Data[string(product)]
+			if buf[0] == buf[1] && buf[0] == buf[2] && buf[0] == buf[3] {
```

Comment:
```suggestion
			if buf != nil && buf[0] == buf[1] && buf[0] == buf[2] && buf[0] == buf[3] {
```
This prevents an index out of bounds error from throwing when performing fresh installation where the `oauth-client-secrets` doesn't already exist

Reply:
@austincunningham I tried pushing a commit to your fork but I don't have the proper permissions. I can lgtm once the change is pushed and e2e tests are passing.

Reply:
thanks carl good catch

Reply:
pushed